File: sdk/ruby/lib/anyvali/parse/coercion_config.rb
# frozen_string_literal: true

module AnyVali
  module CoercionConfig
    PORTABLE_COERCIONS = %w[
      string->int
      string->number
      string->bool
      trim
      lower
      upper
    ].freeze

    def self.portable?(config)
      configs = config.is_a?(Array) ? config : [config]
      configs.all? { |c| PORTABLE_COERCIONS.include?(c) }
    end
  end
end

File: sdk/csharp/src/AnyVali/Parse/CoercionConfig.cs
namespace AnyVali.Parse;

/// <summary>
/// Configuration for value coercion.
/// </summary>
public sealed record CoercionConfig
{
    public string? From { get; init; }
    public bool Trim { get; init; }
    public bool Lower { get; init; }
    public bool Upper { get; init; }
}
